### ISSUE TITLE
Problem with Wolfenstein: Youngblood

### ISSUE THREAD

**Issue by TheSpider12** [emoose/DLSSTweaks#58]
Using DLSSTweaks with the dev dll file (or the latest dll file from TechPowerUp) to inject DLAA into this game results in flickering at the top of the screen. In-game resolution is 5120x2882 (1.78x DLDSR). Below is the content of the dlsstweaks text file generated:

[2023-03-30 14:48:49.967] [info] DLSSTweaks v0.200.5, by emoose: nvngx.dll wrapper loaded
[2023-03-30 14:48:49.967] [info] Game path: C:\Games\Wolfenstein Youngblood\Youngblood_x64vk.exe
[2023-03-30 14:48:49.967] [info] DLL path: C:\Games\Wolfenstein Youngblood\nvngx.dll
[2023-03-30 14:48:49.968] [info] ---
[2023-03-30 14:48:49.968] [info] Config read from C:\Games\Wolfenstein Youngblood\dlsstweaks.ini
[2023-03-30 14:48:49.968] [info] Wrapped nvngx.dll, using funcptrs from original dll
[2023-03-30 14:48:49.968] [debug] LdrRegisterDllNotification callback set
[2023-03-30 14:48:50.201] [debug] nvngx_dlss: applied hud hook via registry
[2023-03-30 14:48:50.358] [info] DLSS functions found & parameter hooks applied!
[2023-03-30 14:48:50.358] [info] Settings:
[2023-03-30 14:48:50.358] [info]  - ForceDLAA: true
[2023-03-30 14:48:50.358] [info]  - OverrideAutoExposure: enable
[2023-03-30 14:48:50.358] [info]  - OverrideAppId: true
[2023-03-30 14:48:50.358] [info]  - OverrideDlssHud: enable
[2023-03-30 14:48:50.358] [info]  - DisableDevWatermark: true
[2023-03-30 14:48:50.358] [info]  - WatchIniUpdates: false
[2023-03-30 14:48:50.358] [info]  - ResolutionOffset: 0
[2023-03-30 14:48:50.358] [info]  - DLSSQualityLevels enabled: false
[2023-03-30 14:48:50.358] [info]  - DLSSPresets:
[2023-03-30 14:48:50.358] [info]   - DLAA: D
[2023-03-30 14:48:50.358] [info]   - Quality: D
[2023-03-30 14:48:50.358] [info]   - Balanced: D
[2023-03-30 14:48:50.358] [info]   - Performance: D
[2023-03-30 14:48:50.358] [info]   - UltraPerformance: D
[2023-03-30 14:48:50.358] [info]  - DLLPathOverrides: N/A

Using the original dll file (2.1.29.0) that comes with the game fixes that flickering and the graphics actually looks almost the same if not better even though it's DLSS instead of DLAA. Also, because DLSS is being used, I can actually turn the resolution up to 5760 x 3240 (2.25x DLDSR) and still get good fps while the image quality improves.

**Comment by emoose** [comment]
Did you try any other presets with the 3.1 DLL? Could be something with the preset it's using, I noticed issues in other games when using certain presets at least (IIRC shadow of tomb raider would show flickering during black screens with preset F, but was fine with other presets)

Also does forcing DLAA work with the original 2.1.29 DLL? AFAIK it should be fine on older versions, but that does look pretty early, so maybe there's some issue with that, didn't really try testing on 2.1.X much iirc.

It could be worth trying one of the later 2.X versions like 2.5.1 instead too, maybe would work better than the 3.1 releases.

**Comment by TheSpider12** [comment]
> Did you try any other presets with the 3.1 DLL? Could be something with the preset it's using, I noticed issues in other games when using certain presets at least (IIRC shadow of tomb raider would show flickering during black screens with preset F, but was fine with other presets)
> 
> Also does forcing DLAA work with the original 2.1.29 DLL? AFAIK it should be fine on older versions, but that does look pretty early, so maybe there's some issue with that, didn't really try testing on 2.1.X much iirc.
> 
> It could be worth trying one of the later 2.X versions like 2.5.1 instead too, maybe would work better than the 3.1 releases.

I tried all the presets with 3.1 dll. All caused the same flickering. But I found the culprit. It's DLDSR. If I go back to native resolution (3840x2160), the flickering is gone and DLAA works as intended. Switching to higher DLDSR resolutions (5120x2882 or 5760 x 3240) brings back the flickering.

Forcing DLAA work with the original 2.1.29 dll, and there's no flickering even at DLDSR resolutions. But performance is worse compared to DLSS + DLDSR. For example, DLAA + DLDSR 1.78x gives me around 100 fps, but DLSS + DLDSR 2.25x gives me a stable 116fps (could be more, as I cap my fps at 116fps). I suppose DLSS + DLDSR 2.25x should look better.

So, to summarize, there may be some compatibility issues going on between DLDSR and the latest dll file that cause the flickering.

A separate issue: I notice that with DLSS or DLAA enabled, whenever I move the character, the objects at far distance become noticeably blurrier (this does not seem to happen in cutscenes). Is this the nature of the tech, or there may be some tweaking we can do in the ini file to deal with that? Enabling DLDSR helps fix the issue somewhat, but I don't think I will be able to use those with the latest more demanding titles.

**Comment by emoose** [comment]
Interesting that DLDSR was the cause, wonder if changing desktop resolution to the DSR res before launching might change anything, maybe the res change during game startup could be messing up the DLSS init somehow.

Do any of the normal DLSS modes (non-DLAA) also flicker when using DLDSR? If it can be made to happen without DLSSTweaks (with just the updated DLSS), that could maybe be worth reporting to NV.
(not really sure how focused they are on DSR any more though, a ton of games released this year that don't seem to work with it properly out-of-the-box at all and need desktop res change first, and we're still waiting on more DLDSR modes to drop...)

Blurry distant objects might be something to do with AutoExposure or the preset you're using, if you use the dev DLL you could try changing both of those during gameplay to see if it helps or not.

(AFAIK AutoExposure does usually give an improvement, but if the game itself was already providing exposure values to DLSS then AutoExposure would override that value, so might end up less accurate and have issues as a result - started adding something that can write into log file if game is providing an exposure value or not, still need to find a title that actually does though, seems like majority of games don't bother with it...)

The blurriness might just be a limitation of DLSS though, maybe the jittering that it needs to work is making them seem blurrier, the dev DLL does let you reduce jitter IIRC but think that might also reduce effectiveness of DLSS/DLAA too.

**Comment by TheSpider12** [comment]
> Interesting that DLDSR was the cause, wonder if changing desktop resolution to the DSR res before launching might change anything, maybe the res change during game startup could be messing up the DLSS init somehow.
> 
> Do any of the normal DLSS modes (non-DLAA) also flicker when using DLDSR? If it can be made to happen without DLSSTweaks (with just the updated DLSS), that could maybe be worth reporting to NV. (not really sure how focused they are on DSR any more though, a ton of games released this year that don't seem to work with it properly out-of-the-box at all and need desktop res change first, and we're still waiting on more DLDSR modes to drop...)
> 
> Blurry distant objects might be something to do with AutoExposure or the preset you're using, if you use the dev DLL you could try changing both of those during gameplay to see if it helps or not.
> 
> (AFAIK AutoExposure does usually give an improvement, but if the game itself was already providing exposure values to DLSS then AutoExposure would override that value, so might end up less accurate and have issues as a result - started adding something that can tell you in the log file if game is providing an exposure value or not, still need to find a title that actually does though, seems like majority of games don't bother with it...)
> 
> The blurriness might just be a limitation of DLSS though, maybe the jittering that it needs to work is making them seem blurrier, the dev DLL does let you reduce jitter IIRC but think that might also reduce effectiveness of DLSS/DLAA too.

After a lot of testing back and forth, I finally found the true reason. It's full res DLAA. After I set ResolutionOffset = -1, the flickering is gone with 3.1 dll. Everything looks fine now.

Now I have to decide between DLAA versus DLSS Quality + DLDSR 2.25x. In theory, both "render" at native res and should give equal performance, but based on Nvidia's claims, DLDSR 2.25x gives the same image quality as DSR 4x, so does it mean DLSS Quality + DLDSR 2.25x is the better choice here image quality wise? If this is the case, isn't DLAA obsolete? What do you think?
